### PR TITLE
server: fix the missing detachment for the mem/disk tracker (#19794)

### LINKED ISSUE #17420: Detach memory tracker for insert/update
## Development Task

Currently, for the session memory tracker, it won't detach from the global memory tracker for `insert` and `update` which would cause the inaccurate memory usage statistical information. We need to detach the memory tracker in these cases.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19794 to release-4.0

---

### What problem does this PR solve?
Currently, we detach session tracker from global tracker in `CloseRecordSet` function. If the query have no `RecordSet` returned (eg: `insert`), the session tracker won't be detached from global tracker which would cause the inaccurate memory usage statistical information.

Issue Number: close #17420 

### What is changed and how it works?

How it Works:

If the ExecuteStmt have no result returned in the `handleQuery`, we will directly detach the tracker 

### Release note
* No release note

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -1535,10 +1535,13 @@ func (e *UnionExec) Close() error {
 // Before every execution, we must clear statement context.
 func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	vars := ctx.GetSessionVars()
+<<<<<<< HEAD
```
Comment (qw4990) [inline executor/executor.go:1538]: Please fix conflicts @Yisaer .

Comment (Yisaer) [reply executor/executor.go]: updated.

**Comment by qw4990** [comment]
Should we add some tests for this? @Yisaer

**Comment by Yisaer** [comment]
As global memory tracker didn't carry into `release-4.0`, I think maybe we don't need to bring the test. @qw4990

**Review by qw4990 — COMMENTED** [review]
LGTM

**Review by SunRunAway — APPROVED** [review]
LGTM

**Pull request merged**